pkg/collector/corechecks/snmp/internal/cache/cache.go

```diff
@@ -0,0 +1,19 @@
+package cache
+
+import "github.com/DataDog/datadog-agent/pkg/persistentcache"
+
+type Cacher interface {
```

Comment:
Is the reason for adding Cacher just for test purposes of being able to use a MockCacher?  If so have you considered using a real persistentcache but with a mocked location for storing it, similar to https://github.com/DataDog/datadog-agent/blob/015ead2d6ad6ed4aa8475a29cc5d794bb8313089/pkg/persistentcache/persistentcache_test.go#L20-L22?

Reply:
Yes it was for the MockCacher !
I am not a big fan of writing in the disk during the tests so I think I prefer to use my interface.
When this PR will be merge I will add an e2e test on this so it will test the persistent cache in any case